core/storage/btree.rs

```diff
@@ -90,6 +99,98 @@ impl BTreeCursor {
         Ok(CursorResult::Ok(page.cell_count() == 0))
     }
 
+    fn get_cell_with<T>(
+        &self,
+        page_idx: usize,
+        cell_idx: usize,
+        handle: impl Fn(BTreeCell) -> Result<CursorResult<T>>,
+    ) -> Result<CursorResult<T>> {
+        self.load_page_with(page_idx, |page| {
+            let cell = page.cell_get(
+                cell_idx,
+                self.pager.clone(),
+                self.max_local(page.page_type()),
+                self.min_local(page.page_type()),
+                self.usable_space(),
+            )?;
+
+            handle(cell)
+        })
+    }
+
+    fn move_to_child_rightmost(&mut self) -> Result<CursorResult<()>> {
+        let mem_page = self.get_mem_page();
+
+        let handle_cell = |cell: BTreeCell| match cell {
```

Comment:
What's the purpose of passing a closure callback to `get_cell_with` and `load_page_with`? Would it work with just something like

```
let cell = self.get_cell_with(...);
/* do whatever handle_cell does, but inline */
```